source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -230,14 +230,15 @@ bool ConnectionManager::ResponseDecoder::onData(Buffer::Instance& data) {
 }
 
 FilterStatus ConnectionManager::ResponseDecoder::transportBegin(MessageMetadataSharedPtr metadata) {
-  return parent_.applyEncoderFilters(DecoderEvent::TransportBegin, metadata, protocol_converter_);
+  parent_.filter_context_ = metadata;
+  return parent_.applyEncoderFilters(DecoderEvent::TransportBegin, protocol_converter_);
```

Comment:
I'm not keen on this pattern of having to correctly set this field before invoking the method. I'm fine substituting FilterContext for absl::any, but I think we should still pass the value in these calls.